app/actions/websocket/scheduled_post.ts

```diff
@@ -6,13 +6,13 @@ import {ActionType} from '@constants';
 import {logError} from '@utils/log';
 
 export type ScheduledPostWebsocketEventPayload = {
-    scheduled_post: string;
+    scheduledPost: string;
 }
 
 export async function handleCreateOrUpdateScheduledPost(serverUrl: string, msg: WebSocketMessage<ScheduledPostWebsocketEventPayload>, prepareRecordsOnly = false) {
     try {
-        const scheduledPost: ScheduledPost = JSON.parse(msg.data.scheduled_post);
-        return handleScheduledPosts(serverUrl, ActionType.SCHEDULED_POSTS.CREATE_OR_UPDATED_SCHEDULED_POST, [scheduledPost], prepareRecordsOnly);
+        const scheduledPost: ScheduledPost[] = msg?.data?.scheduledPost ? [JSON.parse(msg.data.scheduledPost)] : [];
```

Comment:
I think msg is defined in this context. Not so sure about data.

```suggestion
        const scheduledPost: ScheduledPost[] = msg.data?.scheduledPost ? [JSON.parse(msg.data.scheduledPost)] : [];
```

Reply:
Done.